### PR TITLE
design doc for m9

### PR DESCRIPTION
**Pull request by yash2002vardhan**
# Pull Request

## Summary

---

## What Changed

- 
- 

---

## Why This Change?

---

## Testing
- [ ] Tested locally
- [ ] Edge cases considered
- [ ] No breaking changes introduced

### Test Evidence

---

### Areas Affected

- [ ] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [ ] Database
- [ ] NA

### Modules Affected

- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [ ] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [ ] M7 — Hub
- [ ] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

---

## Risks / Concerns

- 

---

## Checklist
- [ ] Code follows project conventions (package layout, no cross-module internal imports)
- [ ] Self-review completed
- [ ] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [ ] No sensitive data or credentials exposed
- [ ] Documentation updated if behaviour changed
- [ ] Ready for review

### CODE REVIEW

File: docs/M9/M9-AIRLINE-PLAN.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    D -->|"No"| E["Look at tomorrow's flights on this route instead"]
+    D -->|"Yes"| F["Estimate the cost of each, using the standard price table"]
+    F --> G["Prefer the cheapest option"]
+    G --> H{"Is a similarly-priced overnight flight available?"}
+    H -->|"Yes"| I["Choose the overnight flight — it's discounted"]
+    H -->|"No"| J["Choose the cheapest option found"]
+    I --> K["This is the flight the batch will use"]
+    J --> K
+```
+
```
Comment (Sidarthapati) [inline docs/M9/M9-AIRLINE-PLAN.md:80]: Have a small doubt here, although i guess we wold have to talk to the airline guy for this, for the cheapest option it would be load wise,so there would be some logic involving which flight already has some weight to it, our goal would be to increase the weight to a certain cutoff. We can add it later but something involving already assigned flights for the same route and the net packet weight inside it (flight bag weight essentially) will come

File: docs/M9/M9-AIRLINE-PLAN.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    E --> F["The batch reaches its cutoff time and is sealed and weighed"]
+    F --> G["We're told the batch's final total weight and package count"]
+    G --> H["We book the whole batch as one flight reservation and calculate the real cost"]
+    H --> I["We receive the full list of packages that were in the batch"]
+    I --> J["Each package is linked to this booking so it can be tracked later"]
+    J --> K["The order-tracking system is told which flight each package is now booked on"]
+```
+
+---
+
```
Comment (Sidarthapati) [inline docs/M9/M9-AIRLINE-PLAN.md:104]: What is this order tracking system, I am not aware we have to make anything like this , I dont think its in any plans, what is that and why do we have to maintain that?  We are already storing current state in m4. Interested to know what order tracking system is.

Also shouldnt you maintain also what parcels are in what bag, and when it reaches cutoff time you should book the airway space and then emit an event and hub will mark it as to be sent to shuttle, curious a to what do you mean were told the batches final weight and package count, i though you were maintaing this

**Review by Sidarthapati — COMMENTED** [review]
LGTM
